index.bs

```diff
@@ -1798,6 +1822,66 @@ Note: Invoking this method from a [=browsing context=] where the [=Web Authentic
 
 </div>
 
+### Client Supported Capabilities (dictionary <dfn dictionary>SupportedCapabilities</dfn>) ### {#dictionary-supportedcapabilities}
+
+<xmp class="idl">
+    dictionary SupportedCapabilities {
+        boolean areU2FDevicesSupported;
+        boolean areCTAPDevicesSupported;
+        boolean areResidentKeysSupported;
+        boolean isUserVerificationSupported;
+        boolean isTokenBindingSupported;
+        UserVerifyingPlatformAuthenticatorAvailability userVerifyingPlatformAuthenticatorAvailability;
+        sequence<AuthenticatorTransport> transportsSupported;
+        sequence<DOMString> extensionsSupported;
+    };
+</xmp>
+
+This dictionary is used by [=[WRPS]=] to determine the supported capabilities of the [=client=].
+
+<div dfn-type="dict-member" dfn-for="SupportedCapabilities">
+    :   <dfn>areU2FDevicesSupported</dfn>
+    ::  This attribute indicates whether the [=client=] supports communicating with [[FIDO-U2F-Message-Formats]] devices.
+
+    :   <dfn>areCTAPDevicesSupported</dfn>
+    ::  This attribute indicates whether the [=client=] supports communicating with [[FIDO-CTAP]] devices.
+
+    :   <dfn>areResidentKeysSupported</dfn>
+    ::  This attribute indicates whether the [=client=] supports creating [=resident credentials=].
+
+    :   <dfn>isUserVerificationSupported</dfn>
```

Comment:
Is this different from isUserVerifyingPlatformAuthenticatorAvailable()?

Reply:
isUserVerifyingPlatformAuthenticatorAvailable currently indicates the enabled state of the UVPA, but the RP cannot distinguish between disabled/unavailable.  I left isUserVerifyingPlatformAuthenticatorAvailable alone as RPs are currently using this API, but reflected the tri-state in this new API.

Reply:
I was asking about the purpose of isUserVerificationSupported() function. We have userVerifyingPlatformAuthenticatorAvailability() and isUVPAA(). What is the difference?

Reply:
Thanks for clarifying.  The intention of isUserVerificationSupported is so that the client can covey whether it supports the UserVerificationRequirement enum.  On the other hand, userVerifyingPlatformAuthenticatorAvailability is intended to indicate the specific state of the UVPA on the client platform.  Does that make sense?

Reply:
Make sense.
So, this method is for indicating the client supports userVerification option. IMO, it is better to refer [userVerification](https://w3c.github.io/webauthn/#dom-authenticatorselectioncriteria-userverification) member instead.

Reply:
I suggest that the for this (and all other similar functionality detection), we remove these methods and say that if a particular capability is not part of  the response from `getSupportedCapabilities()`, then it's not supported.

Reply:
If I understand the comment of saying to remove isUserVerifyingPlatformAuthenticatorAvailable(), browsers already implement this function and RPs already depend on it.  I don't think we should remove it.

Reply:
I would prefer the way suggested by @leshi . WebAuthn spec will keep introducing new features. Returning the supported features as a list of string is better and easy to extend.

Reply:
yes, IIUC, whatever we do, we need to leave `isUserVerifyingPlatformAuthenticatorAvailable()` nominally intact, and if we do decide to deprecate it in light of other new functionality we define, we ought to mark it (in the spec) as deprecated and point to the new functionality.

Reply:
I have removed references to an updated UVPAA check, as we currently don't have scenarios to leverage this additional granularity.  I have also adopted the style used by Transports to return the list of supported capabilities.